### DISCUSSION TITLE
%20 in query string being stripped out

### DISCUSSION THREAD

**Discussion by agricolan** [YOURLS/YOURLS#2939] [Q&A] [answered]
I'm posting a URL with a space in the query:
https://www.reach22.com/p/yourls-api.php?signature=xxxxxxxxxx&action=shorturl&keyword=test&format=json&url=https://reach22.com?title=coffee%20cup
I receive the response:
"status":"success","message":"https:\/\/reach22.com\/coffeecup added to database"

How can I persuade YOURLS to retain the space in coffee cup.  Works fine with the Admin Interface but not with the API. Thanks

**Comment by agricolan** [comment]
Managed to fix it by changing %20 to \\%20.

**Comment by agricolan** [comment]
That should be a double backslash.

**Comment by ozh** [comment, marked as answer]
As stated in the readme: URL encode your requests. http://yourls.org/#API;apiusage

**Comment by agricolan** [comment]
Tnanks.  In Python 3 I'm using:

url = urllib.parse.quote_plus(sys.argv[1]) where sys.argv[1]="coffee 
cup" but it doesn't work  - the youurl has 'coffeecup' and not 'coffee 
cup'.

Is there another command I should be using?

Michael

On 2021-05-15 18:37, ྅༻ Ǭɀħ ༄༆ཉ wrote:

> As stated in the readme: URL encode your requests. 
> http://yourls.org/#API;apiusage
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].


Links:
------
[1] 
https://github.com/YOURLS/YOURLS/discussions/2939#discussioncomment-742399
[2]

**Reply by dgw** [reply]
At least I would say use `quote()` instead, since `quote_plus()` explicitly doesn't convert space to `%20` but rather to `+`.

That said, if you're using a library like `requests` to send your API calls, you should just [let the library handle](https://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls) your encoding for you.
